@scholzj This is the output when i run `mvn checkstyle:check`
 There are 21 errors reported by Checkstyle 9.0 with sun_checks.xml ruleset.


``` There are 21 errors reported by Checkstyle 9.0 with sun_checks.xml ruleset.
[ERROR] src/main/java/io/strimzi/common/ConfigUtil.java:[1] (javadoc) JavadocPackage: Missing package-info.java file.
[ERROR] src/main/java/io/strimzi/common/ConfigUtil.java:[3] (sizes) LineLength: Line is longer than 80 characters (found 101).
[ERROR] src/main/java/io/strimzi/common/ConfigUtil.java:[10,1] (design) HideUtilityClassConstructor: Utility classes should not have a public or default constructor.
[ERROR] src/main/java/io/strimzi/common/ConfigUtil.java:[14] (sizes) LineLength: Line is longer than 80 characters (found 93).
[ERROR] src/main/java/io/strimzi/common/ConfigUtil.java:[15] (sizes) LineLength: Line is longer than 80 characters (found 120).
[ERROR] src/main/java/io/strimzi/common/ConfigUtil.java:[17,53] (misc) FinalParameters: Parameter envVar should be final.
[ERROR] src/main/java/io/strimzi/common/ConfigUtil.java:[18] (sizes) LineLength: Line is longer than 80 characters (found 89).
[ERROR] src/main/java/io/strimzi/common/TracingInitializer.java:[3] (sizes) LineLength: Line is longer than 80 characters (found 101).
[ERROR] src/main/java/io/strimzi/common/TracingInitializer.java:[12] (javadoc) JavadocStyle: First sentence should end with a period.
[ERROR] src/main/java/io/strimzi/common/TracingInitializer.java:[15,1] (design) HideUtilityClassConstructor: Utility classes should not have a public or default constructor.
[ERROR] src/main/java/io/strimzi/common/TracingInitializer.java:[16] (javadoc) JavadocStyle: First sentence should end with a period.
[ERROR] src/main/java/io/strimzi/common/TracingInitializer.java:[17] (sizes) LineLength: Line is longer than 80 characters (found 83).
[ERROR] src/main/java/io/strimzi/common/TracingInitializer.java:[20] (sizes) LineLength: Line is longer than 80 characters (found 88).
[ERROR] src/main/java/io/strimzi/common/TracingInitializer.java:[29] (sizes) LineLength: Line is longer than 80 characters (found 90).
[ERROR] src/main/java/io/strimzi/common/TracingSystem.java:[3] (sizes) LineLength: Line is longer than 80 characters (found 101).
[ERROR] src/main/java/io/strimzi/common/TracingSystem.java:[7] (javadoc) JavadocStyle: First sentence should end with a period.
[ERROR] src/main/java/io/strimzi/common/TracingSystem.java:[11,5] (javadoc) JavadocVariable: Missing a Javadoc comment.
[ERROR] src/main/java/io/strimzi/common/TracingSystem.java:[12,5] (javadoc) JavadocVariable: Missing a Javadoc comment.
[ERROR] src/main/java/io/strimzi/common/TracingSystem.java:[13,5] (javadoc) JavadocVariable: Missing a Javadoc comment.
[ERROR] src/main/java/io/strimzi/common/TracingSystem.java:[15,5] (javadoc) MissingJavadocMethod: Missing a Javadoc comment.
[ERROR] src/main/java/io/strimzi/common/TracingSystem.java:[15,42] (misc) FinalParameters: Parameter value should be final.```

--- a/.checkstyle/checkstyle.xml
+++ b/.checkstyle/checkstyle.xml
@@ -94,32 +94,21 @@
         <module name="MethodLength"/>
         <module name="ParameterNumber">
             <!-- default is 8 -->
-            <property name="max" value="13"/>
+            <property name="max" value="11"/>
         </module>
         <module name="ClassDataAbstractionCoupling">
             <!-- default is 7 -->
-            <property name="max" value="20"/>
+            <property name="max" value="9"/>
         </module>
-        <module name="BooleanExpressionComplexity">
-            <!-- default is 3 -->
-            <property name="max" value="5"/>
-        </module>
-
+        <module name="BooleanExpressionComplexity"/>
         <module name="ClassFanOutComplexity">
             <!-- default is 20 -->
-            <property name="max" value="44"/>
+            <property name="max" value="21"/>
         </module>
-        <module name="CyclomaticComplexity">
-            <!-- default is 10-->
-            <property name="max" value="19"/>
-        </module>
-        <module name="JavaNCSS">
-            <!-- default is 50 -->
-            <property name="methodMaximum" value="100"/>
-        </module>
+        <module name="CyclomaticComplexity"/>
+        <module name="JavaNCSS"/>
         <module name="NPathComplexity">
-            <!-- default is 200 -->
-            <property name="max" value="5832"/>
+            <property name="max" value="256"/>
         </module>
 
         <module name="IllegalToken">

--- a/.checkstyle/checkstyle.xml
+++ b/.checkstyle/checkstyle.xml
@@ -3,7 +3,7 @@
         "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
         "http://checkstyle.sourceforge.net/dtds/configuration_1_3.dtd">
 
-<module name = "Checker">
+<module name="Checker">
 
     <property name="localeLanguage" value="en"/>
 
@@ -11,14 +11,18 @@
 
     <!-- header -->
     <module name="RegexpHeader">
-        <property name="headerFile" value="${checkstyle.header.file}"/>
+        <!--
+            /*
+             * Copyright Strimzi authors.
+             * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+             */
+         -->
+        <property name="header" value="^\/\*$\n^\s\*\sCopyright\sStrimzi\sauthors\.$\n^\s\*\sLicense:\sApache\sLicense\s2\.0\s\(see\sthe\sfile\sLICENSE\sor\shttp:\/\/apache\.org\/licenses\/LICENSE-2\.0\.html\)\.$\n^\s\*\/$"/>
         <property name="fileExtensions" value="java"/>
     </module>
 
     <module name="SuppressWarningsFilter" />
-
     <module name="TreeWalker">
-
         <!-- code cleanup -->
         <module name="UnusedImports">
             <property name="processJavadoc" value="true" />
@@ -90,22 +94,24 @@
         <module name="MethodLength"/>
         <module name="ParameterNumber">
             <!-- default is 8 -->
+            <property name="max" value="13"/>
         </module>
         <module name="ClassDataAbstractionCoupling">
             <!-- default is 7 -->
-            <property name="max" value="12"/>
+            <property name="max" value="20"/>
         </module>
         <module name="BooleanExpressionComplexity">
             <!-- default is 3 -->
+            <property name="max" value="5"/>
         </module>
 
         <module name="ClassFanOutComplexity">
             <!-- default is 20 -->
-            <property name="max" value="36"/>
+            <property name="max" value="44"/>
         </module>
         <module name="CyclomaticComplexity">
             <!-- default is 10-->
-            <property name="max" value="14"/>
+            <property name="max" value="19"/>
         </module>
         <module name="JavaNCSS">
             <!-- default is 50 -->
@@ -113,6 +119,7 @@
         </module>
         <module name="NPathComplexity">
             <!-- default is 200 -->
+            <property name="max" value="5832"/>
         </module>
 
         <module name="IllegalToken">

--- a/java/common/src/main/java/io/strimzi/common/ConfigUtil.java
+++ b/java/common/src/main/java/io/strimzi/common/ConfigUtil.java
@@ -4,6 +4,8 @@
  */
 package io.strimzi.common;
 
+import java.util.Locale;
+
 /**
  * Provides utility methods for managing common configuration properties.
  */
@@ -15,6 +17,6 @@ public class ConfigUtil {
      * @return Returns a String which removes a prefix containing '_', converts to lower case and replaces '_' with '.'.
      */
     public static String convertEnvVarToPropertyKey(String envVar) {
-        return envVar.substring(envVar.indexOf("_") + 1).toLowerCase().replace("_", ".");
+        return envVar.substring(envVar.indexOf("_") + 1).toLowerCase(Locale.ENGLISH).replace("_", ".");
     }
 }

--- a/java/http/java-http-producer/src/main/java/HttpProducer.java
+++ b/java/http/java-http-producer/src/main/java/HttpProducer.java
@@ -60,7 +60,7 @@ public class HttpProducer {
         messagesSentLatch.await();
     }
 
-    public HttpProducer(HttpProducerConfig config,CountDownLatch messagesSentLatch ) throws URISyntaxException {
+    public HttpProducer(HttpProducerConfig config, CountDownLatch messagesSentLatch) throws URISyntaxException {
         this.config = config;
         this.messagesSentLatch = messagesSentLatch;
         this.executorService = Executors.newSingleThreadScheduledExecutor();

--- a/java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java
+++ b/java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumer.java
@@ -83,7 +83,7 @@ public class HttpKafkaConsumer extends AbstractVerticle {
             startPromise.complete();
             return Future.succeededFuture();
         })
-        .onFailure(startPromise::fail);
+            .onFailure(startPromise::fail);
     }
 
     @Override

--- a/java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumerConfig.java
+++ b/java/http/vertx/java-http-vertx-consumer/src/main/java/HttpKafkaConsumerConfig.java
@@ -175,7 +175,7 @@ public class HttpKafkaConsumerConfig {
         boolean pipelining = Boolean.valueOf(map.getOrDefault(ENV_PIPELINING, DEFAULT_PIPELINING).toString());
         int pipeliningLimit = Integer.parseInt(map.getOrDefault(ENV_PIPELINING_LIMIT, DEFAULT_PIPELINING_LIMIT).toString());
         String envMessageCount = (String) map.get(ENV_MESSAGE_COUNT);
-        Optional<Long> messageCount = envMessageCount != null ? Optional.of((Long.parseLong(envMessageCount))) : Optional.empty();
+        Optional<Long> messageCount = envMessageCount != null ? Optional.of(Long.parseLong(envMessageCount)) : Optional.empty();
         String endpointPrefix = (String) map.getOrDefault(ENV_ENDPOINT_PREFIX, DEFAULT_ENDPOINT_PREFIX);
         return new HttpKafkaConsumerConfig(hostname, port, topic, clientId, groupid, pollInterval, pollTimeout, pipelining, pipeliningLimit, messageCount, endpointPrefix);
     }

--- a/java/http/vertx/java-http-vertx-producer/src/main/java/HttpKafkaProducer.java
+++ b/java/http/vertx/java-http-vertx-producer/src/main/java/HttpKafkaProducer.java
@@ -133,7 +133,7 @@ public class HttpKafkaProducer extends AbstractVerticle {
 
                     if (this.config.getMessageCount().isPresent() &&
                         this.messagesSent >= this.config.getMessageCount().get()) {
-                            // signal to main thread that all messages are sent, application can exit
+                        // signal to main thread that all messages are sent, application can exit
                         this.messagesSentLatch.countDown();
                         log.info("All messages sent");
                     }

--- a/java/http/vertx/java-http-vertx-producer/src/main/java/HttpKafkaProducer.java
+++ b/java/http/vertx/java-http-vertx-producer/src/main/java/HttpKafkaProducer.java
@@ -86,7 +86,7 @@ public class HttpKafkaProducer extends AbstractVerticle {
         stopPromise.complete();
     }
 
-    private Promise <List<OffsetRecordSent>> send(String topic, Span span) {
+    private Promise<List<OffsetRecordSent>> send(String topic, Span span) {
         Promise<List<OffsetRecordSent>> fut = Promise.promise();
 
         JsonObject records = new JsonObject();
@@ -121,7 +121,7 @@ public class HttpKafkaProducer extends AbstractVerticle {
                                 list.add(new OffsetRecordSent(
                                     json.getInteger("partition"),
                                     json.getLong("offset"))
-                                    );
+                                );
                             });
                             fut.complete(list);
                         } else {
@@ -134,8 +134,8 @@ public class HttpKafkaProducer extends AbstractVerticle {
                     if (this.config.getMessageCount().isPresent() &&
                         this.messagesSent >= this.config.getMessageCount().get()) {
                             // signal to main thread that all messages are sent, application can exit
-                            this.messagesSentLatch.countDown();
-                            log.info("All messages sent");
+                        this.messagesSentLatch.countDown();
+                        log.info("All messages sent");
                     }
                 });
         return fut;

--- a/java/http/vertx/java-http-vertx-producer/src/main/java/HttpKafkaProducerConfig.java
+++ b/java/http/vertx/java-http-vertx-producer/src/main/java/HttpKafkaProducerConfig.java
@@ -108,7 +108,7 @@ public class HttpKafkaProducerConfig {
         String topic = (String) map.getOrDefault(ENV_TOPIC, DEFAULT_TOPIC);
         int sendInterval = Integer.parseInt(map.getOrDefault(ENV_SEND_INTERVAL, DEFAULT_SEND_INTERVAL).toString());
         String envMessageCount = (String) map.get(ENV_MESSAGE_COUNT);
-        Optional<Long> messageCount = envMessageCount != null ? Optional.of((Long.parseLong(envMessageCount))) : Optional.empty();
+        Optional<Long> messageCount = envMessageCount != null ? Optional.of(Long.parseLong(envMessageCount)) : Optional.empty();
         String endpointPrefix = (String) map.getOrDefault(ENV_ENDPOINT_PREFIX, DEFAULT_ENDPOINT_PREFIX);
         return new HttpKafkaProducerConfig(hostname, port, topic, sendInterval, messageCount, endpointPrefix);
     }

--- a/java/kafka/streams/src/main/java/io/strimzi/kafka/streams/KafkaStreamsConfig.java
+++ b/java/kafka/streams/src/main/java/io/strimzi/kafka/streams/KafkaStreamsConfig.java
@@ -38,7 +38,7 @@ public class KafkaStreamsConfig {
                 .stream()
                 .filter(mapEntry -> mapEntry.getKey().startsWith(KAFKA_PREFIX))
                 .collect(Collectors.toMap(mapEntry -> ConfigUtil.convertEnvVarToPropertyKey(mapEntry.getKey()), Map.Entry::getValue)));
-        return new KafkaStreamsConfig( sourceTopic, targetTopic, tracingSystem, properties);
+        return new KafkaStreamsConfig(sourceTopic, targetTopic, tracingSystem, properties);
     }
 
     public String getSourceTopic() {

--- a/pom.xml
+++ b/pom.xml
@@ -26,6 +26,9 @@
         <maven.shade.version>3.4.1</maven.shade.version>
         <maven.checkstyle.version>3.2.0</maven.checkstyle.version>
 
+        <!-- Build tools -->
+        <checkstyle.version>9.2.1</checkstyle.version>
+
         <!-- Regular dependencies -->
         <kafka.version>3.4.0</kafka.version>
         <log4j.version>2.17.1</log4j.version>
@@ -179,36 +182,6 @@
             <plugins>
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
-                    <artifactId>maven-checkstyle-plugin</artifactId>
-                    <version>${maven.checkstyle.version}</version>
-                    <dependencies>
-                        <dependency>
-                            <groupId>com.puppycrawl.tools</groupId>
-                            <artifactId>checkstyle</artifactId>
-                            <version>9.0</version>
-                        </dependency>
-                    </dependencies>
-                    <executions>
-                        <execution>
-                            <id>validate</id>
-                            <phase>validate</phase>
-                            <configuration>
-                                <configLocation>.checkstyle/checkstyle.xml</configLocation>
-                                <headerLocation>.checkstyle/java.header</headerLocation>
-                                <suppressionsLocation>.checkstyle/suppressions.xml</suppressionsLocation>
-                                <includeTestSourceDirectory>true</includeTestSourceDirectory>
-<!--                                <encoding>UTF-8</encoding>-->
-                                <consoleOutput>true</consoleOutput>
-                                <failsOnError>true</failsOnError>
-                            </configuration>
-                            <goals>
-                                <goal>check</goal>
-                            </goals>
-                        </execution>
-                    </executions>
-                </plugin>
-                <plugin>
-                    <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-compiler-plugin</artifactId>
                     <version>${maven.compiler.version}</version>
                     <executions>
@@ -255,5 +228,38 @@
                 </plugin>
             </plugins>
         </pluginManagement>
+
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-checkstyle-plugin</artifactId>
+                <version>${maven.checkstyle.version}</version>
+                <dependencies>
+                    <dependency>
+                        <groupId>com.puppycrawl.tools</groupId>
+                        <artifactId>checkstyle</artifactId>
+                        <version>${checkstyle.version}</version>
+                    </dependency>
+                </dependencies>
+                <executions>
+                    <execution>
+                        <id>validate</id>
+                        <phase>validate</phase>
+                        <configuration>
+                            <configLocation>.checkstyle/checkstyle.xml</configLocation>
+                            <headerLocation>.checkstyle/java.header</headerLocation>
+                            <suppressionsLocation>.checkstyle/suppressions.xml</suppressionsLocation>
+                            <includeTestSourceDirectory>true</includeTestSourceDirectory>
+                            <encoding>UTF-8</encoding>
+                            <consoleOutput>true</consoleOutput>
+                            <failsOnError>true</failsOnError>
+                        </configuration>
+                        <goals>
+                            <goal>check</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
     </build>
 </project>

